src/multipole_header.F90

```diff
@@ -0,0 +1,152 @@
+  integer :: max_L     ! Maximum L value for allocation
+
+!===============================================================================
+! MULTIPOLE contains all the components needed for the windowed multipole
+! temperature dependent cross section libraries for the resolved resonance
+! region.
+!===============================================================================
+
+  type MultipoleArray
+
+    !=========================================================================
+    ! Isotope Properties
+    logical                 :: fissionable = .false.  ! Is this isotope fissionable?
+    integer                 :: length                 ! Number of poles
+    integer, allocatable    :: l_value(:)             ! The l index of the pole
+    real(8), allocatable    :: pseudo_k0RS(:)         ! pseudo_k0RS for each of l
+    complex(8), allocatable :: data(:,:)              ! Contains all of the pole-residue data
+    real(8)                 :: sqrtAWR                ! Square root of the atomic weight ratio
+
+    !=========================================================================
+    ! Windows
+
+    integer :: windows                      ! Number of windows
+    integer :: fit_order                    ! Order of the fit. 1 linear, 2 quadratic, etc.
+    real(8) :: start_E                      ! Start energy for the windows
+    real(8) :: end_E                        ! End energy for the windows
+    real(8) :: spacing                      ! The actual spacing in sqrt(E) space.
+    ! spacing = sqrt(multipole_w%endE - multipole_w%startE)/multipole_w%windows
+    integer, allocatable :: w_start(:)      ! Contains the index of the pole at the start of the window
+    integer, allocatable :: w_end(:)        ! Contains the index of the pole at the end of the window
+    real(8), allocatable :: curvefit(:,:,:) ! Contains the fitting function.  (reaction type, coeff index, window index)
+
+    integer, allocatable :: broaden_poly(:) ! if 1, broaden, if 0, don't.
+
+    !=========================================================================
+    ! Storage Helpers
+    integer :: num_l
+    integer :: max_w
+
+    integer :: formalism
+
+    contains
+      procedure :: clear => multipole_clear ! Deallocates Multipole
+      procedure :: allocate => multipole_allocate ! Allocates Multipole
+  end type MultipoleArray
+
+contains
+
+!===============================================================================
+! MULTIPOLE_CLEAR resets and deallocates data in Multipole.
+!===============================================================================
+
+    subroutine multipole_clear(this)
+      class(MultipoleArray), intent(inout) :: this
+
+      if (allocated(this % data)) then
+        deallocate(this % data)
+        deallocate(this % w_start)
+        deallocate(this % w_end)
+        deallocate(this % curvefit)
+
+        deallocate(this % l_value)
+        deallocate(this % pseudo_k0RS)
+      end if
+    end subroutine
```

Comment:
This subroutine can be removed altogether. When the array of multipole data is deallocated, all components that are `allocatable` are automatically deallocated. We really only need these "clear" methods if there are `pointer`s within a derived type.